.github/workflows/tinygo-extract-version.sh

```diff
@@ -0,0 +1,4 @@
+#!/bin/sh
+
+# Extract the version string from the source code, to be stored in a variable.
+grep 'const version' goenv/version.go | sed 's/^const version = "\(.*\)"$/version=\1/g'
```

Comment:
grep not needed
```
sed -n '/^const version =/{ s/^const version = "\(.*\)"$/version=\1/; p;}' goenv/version.go
```

Reply:
True, though I think the `grep` and `sed` together are more readable? This way the `sed` is just a simple regex search-and-replace instead of the more complicated code you wrote that I don't quite understand.
I can update if you think avoiding `grep` is better.

Reply:
Nah, you can leave the grep if you prefer it.

Here's a breakdown of the `sed` code.

Sed, like awk, is can do `/pattern/ { action }`, so that's what we have here.  The `-n` flag tells `sed` not to print every line (the default), so  in our action after the replacement we need to explicitly `p` to print the current line.  If this was happening on a huge file I'd probably add a `q` to quit so `sed` didn't continue to process the entire file after it found the one match it was looking for